Remotion/Web/Core/Resources/IResourcePathBuilder.cs

```diff
@@ -39,5 +39,7 @@ namespace Remotion.Web.Resources
  public interface IResourcePathBuilder
   {
     string BuildAbsolutePath (Assembly assembly, params string[] assemblyRelativePathParts);
+
+    string BuildCacheableAbsolutePath (Assembly assembly, params string[] assemblyRelativePathParts);
```

Comment:
Got more detailed idea:

since we can handle / convert `null` to dot (`"."`) anyway, we can make the `IStaticResourceCacheKeyProvider` part of the ctor signature of the `ResourcePathBuilder` and have just two derived types required for IoC only. One type is the cache-enbled one and accepts the `IStaticResourceCacheKeyProvider`, the other one uses the NullStaticResourceCacheKeyProvider when performing the base ctor call. That way we can register two different `ResourcePathBuilder` for two different `IResourcePathBuilder` interfaces but don't have any implementation differences and thus keep things simple. Should we one day decide to support IoC  with tagged services, this can be further simplified.

Disclaimer: this comment should only simplify the orignal review remarks :)